File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    def dfs(index, sum):
        nonlocal answer
        if index == len(numbers):
            if sum == target:
                answer += 1
            return
        dfs(index + 1, sum + numbers[index])
        dfs(index + 1, sum - numbers[index])
    dfs(0, 0)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js


function solution(numbers, target) {
    let answer = 0;
    function DFS(index, sum) {
        if (index === numbers.length) {
            if (sum === target) {
                answer += 1;
            }
            return;
        } 
        DFS(index + 1, sum + numbers[index]);
        DFS(index + 1, sum - numbers[index]);
    }
    DFS(0, 0)
    return answer;
}